src/WebJobs.Extensions.DurableTask/ProviderUtils.cs

```diff
@@ -16,6 +19,22 @@ namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
     /// </summary>
     public static class ProviderUtils
     {
+#if !FUNCTIONS_V1
+        /// <summary>
+        /// Providers can implement this interface to override the autoscaling.
+        /// </summary>
+        public interface IProviderWithAutoScaling
+        {
+            /// <summary>
+            /// Tries to obtain a scale monitor for autoscaling.
+            /// </summary>
+            /// <param name="extension">The extension, which is used for tracing.</param>
+            /// <param name="scaleMonitor">The scale monitor.</param>
+            /// <returns>True if autoscaling is supported, false otherwise.</returns>
+            bool TryGetScaleMonitor(DurableTaskExtension extension, out IScaleMonitor scaleMonitor);
```

Comment:
I notice that the Azure Storage implementation of IScaleMonitor takes in a FunctionId and a FunctionName to generate a scale monitor descriptor. I'm not sure how necessary it is to have a unique ScaleMonitorDescriptor per trigger, especially if all Durable Triggers for an app should use similar scale logic, but if we do need that, we should put those as parameters on this method.